### PR TITLE
chore: fix temporarily skipped specs on 10.0-release

### PR DESCRIPTION
**Pull request by lmiller1990**
I also fixed some more here skipped ones here https://github.com/cypress-io/cypress/commit/424ae9ac2462c008138e49a5800f4d201c3c932f, not sure how I managed to push to 10.0-release (protected branch) but I accidentally did.

All the tests that were temporarily skipped before xmas are now green again.

### CODE REVIEW

File: system-tests/test/plugins_spec.js [hunk 1]
```diff
@@ -217,18 +211,5 @@ describe('e2e plugins', function () {
         },
       })
     })
-
```
Comment (lmiller1990) [inline system-tests/test/plugins_spec.js:220]: This is no longer a consideration, config-file: false is getting deprecated

https://cypressio.slack.com/archives/C02B9B78H1P/p1641542070002100

File: packages/data-context/src/DataContext.ts [hunk 2]
```diff
@@ -349,8 +349,11 @@ export class DataContext {
 
   onError = (err: Error) => {
     if (this.isRunMode) {
-      // console.error(err)
-      throw err
+      if (this.lifecycleManager?.runModeExitEarly) {
```
Comment (lmiller1990) [inline packages/data-context/src/DataContext.ts:352]: This is a bit of a hack, when we implementing run mode for unified (next week) we can probably start to make this control flow less convoluted.

**Comment by emilyrohrbough** [comment]
@lmiller1990 these tests have also been skipped in the server, will they be fixed in this PR?
- [packages/server/test/unit/plugins/child/run_plugins_spec.js](https://github.com/cypress-io/cypress/blob/10.0-release/packages/server/test/unit/plugins/child/run_plugins_spec.js#L25) - (I have a few change on my local I could share if you haven't started this)
- [packages/server/test/unit/plugins/child/preprocessor_spec.js](https://github.com/cypress-io/cypress/blob/10.0-release/packages/server/test/unit/plugins/child/preprocessor_spec.js)

**Review by tgriesser — APPROVED** [review]
Nice, thanks for taking care of this!

**Pull request merged**